OpenLaTeXStudio/EditorModule/src/latexstudio/editor/EditorTopComponent.java

```diff
@@ -291,13 +309,51 @@ void writeProperties(java.util.Properties p) {
         // http://wiki.apidesign.org/wiki/PropertyFiles
         p.setProperty("version", "1.0");
         // TODO store your settings
-    }
-
+        if(currentFile != null) {
+           p.setProperty(CURRENT_FILE_KEY, currentFile.getAbsolutePath());
+        }
+        else {
+            p.remove(CURRENT_FILE_KEY);
+        }
+        if(latexPath != null) {
+           p.setProperty(LATEX_PATH_KEY, latexPath);
+        }
+        else {
+            p.remove(LATEX_PATH_KEY);
+        }
+        PropertyService.writeProperties(p);
+    } 
+   //previous argumnent: java.util.Properties p
     void readProperties(java.util.Properties p) {
-        String version = p.getProperty("version");
         // TODO read your settings according to their version
+        String currentFilePath;
+        File candidateFile = null;
+        p = PropertyService.readProperties(PropertyService.getPropertyFilePath());
+        if(p == null)
+        {
```

Comment:
As per our project and Java coding convention, the opening brace should be in the same line as method signature or instruction. This problem appears in several places in the code.